ocaml/xapi/cli_operations.ml

```diff
@@ -2287,6 +2287,22 @@ let vm_pause printer rpc session_id params =
 let vm_unpause printer rpc session_id params =
 	ignore(do_vm_op printer rpc session_id (fun vm -> Client.VM.unpause rpc session_id (vm.getref ())) params [])
 
+(* A helper function for VM install *)
+let is_recommended recommendations_xml fieldname =
+	let rec seek_recommendation i =
+		if Xmlm.eoi i then false
+		else match Xmlm.input i with
+			| `El_start ((ns, tag), attrs) when tag = "restriction" && (List.mem ((ns, "field"), fieldname) attrs) ->
+				List.assoc (ns, "value") attrs = "true"
```

Comment:
Ah, I see we're matching on the fieldname, so as long as we don't try to use this function with a restriction field name that doesn't have a `value` attribute, we should be OK.